engine/lite/components/request-response/src/main/scala/pl/touk/nussknacker/engine/lite/components/requestresponse/jsonschema/jsonschemautils/JsonRequestResponseBaseTransformer.scala

```diff
@@ -0,0 +1,43 @@
+package pl.touk.nussknacker.engine.lite.components.requestresponse.jsonschema.jsonschemautils
+
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import org.everit.json.schema.Schema
+import pl.touk.nussknacker.engine.api.{MetaData, NodeId}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.CustomNodeError
+import pl.touk.nussknacker.engine.api.context.transformation.{NodeDependencyValue, SingleInputGenericNodeTransformation}
+import pl.touk.nussknacker.engine.api.definition.{NodeDependency, TypedNodeDependency}
+import pl.touk.nussknacker.engine.api.process.SinkFactory
+
+import scala.util.{Failure, Success, Try}
+
+trait JsonRequestResponseBaseTransformer[T] extends SingleInputGenericNodeTransformation[T] with SinkFactory {
```

Comment:
I think we don't need inheritance here. It can be just a class, like: `SchemaExtractor` with one method getSchemaFromProperty

Reply:
Hmm... It could make it more usable in the future. I have created simple class `JsonSchemaExtractor` with just `getSchemaFromProperty` and moved `nodeDependencies` to sink/source classes directly.